telemetry/packages/constants/src/pods/states.ts

```diff
@@ -29,3 +29,26 @@ export const ALL_POD_STATES = {
   ...ACTIVE_STATES,
   ...NULL_STATES,
 };
+
+export const ALL_POD_STATE_TYPES = [
+  'FAILURE',
+  'PASSIVE',
+  'ACTIVE',
+  'NULL',
+] as const;
+
+export const getStateType = (
+  state: string,
+): (typeof ALL_POD_STATE_TYPES)[number] => {
+  if (FAILURE_STATES[state as keyof typeof FAILURE_STATES]) {
+    return 'FAILURE';
+  } else if (PASSIVE_STATES[state as keyof typeof PASSIVE_STATES]) {
```

Comment:
```suggestion
  }
  
  if (PASSIVE_STATES[state as keyof typeof PASSIVE_STATES]) {
```

You don't need the `else if`s here because you're returning (i.e. just early return! 😄)

Reply:
Haha quite right - copilot wrote this, should've checked it 😂